### PR TITLE
Fix (probably) harmless bad memory accesses in Hexagon simulator runtime

### PR DESCRIPTION
**Pull request by dsharletg**

### CODE REVIEW

**Comment by dsharletg** [comment]
Actually, I'm going to make a different fix.

**Comment by dsharletg** [comment]
Bah never mind, there really isn't a great way to do this. I *really* wish HexagonWrapper had a simple memcpy-like API instead of these insane ReadMemory/WriteMemory functions.

File: src/runtime/hexagon_remote/sim_host.cpp [hunk 1]
```diff
@@ -109,19 +109,30 @@ int write_memory(int dest, const void *src, int size) {
         // WriteMemory only works with powers of 2, so align down. It
         // also only writes up to 8 bytes, so we need to do this
         // repeatedly until we've finished copying the buffer.
-        int next = 1;
-        if (size >= 8) next = 8;
-        else if (size >= 4) next = 4;
-        else if (size >= 2) next = 2;
-        HEXAPI_Status status = sim->WriteMemory(dest, next, *reinterpret_cast<const HEX_8u_t*>(src));
+        HEX_8u_t src_next;
```
Comment (zvookin) [inline src/runtime/hexagon_remote/sim_host.cpp]: I think this code would be easier to read if src_next was named src_chunk. I believe it holds the actual bits, not a pointer to the bits. src_next implies the next value of src and src is a pointer. I'd also name "size" to "remaining_size" and "size_next" to "chunk_size". (To me size_next implies it is the size to use on the next iteration of the loop.) Other naming is possible, and this is surely not a major issue.

**Comment by dsharletg** [comment]
I accidentally rolled the renaming suggestions by @zvookin into the merge commit... :)

**Comment by zvookin** [comment]
Sweet. LGTM.

**Pull request merged**